lec11-modelling.Rmd

```diff
@@ -0,0 +1,527 @@
+standard deviation of 1). And yet, there are still ~6 or so (below the line)
+tests that show a "significant" difference between x and y! Which makes sense,
+since a p-value of 0.05 equals a 5% chance you will see another similar result
+if the null hypothesis were true. Confusing eh?
+
+#### Challenge
+
+1. Play around with the above code and try this different conditions. What do
+you notice?
+    - `sample_size` to 10
+    - `sample_size` to 500
+    - `num_tests` to 1000
+    - `num_tests` to 10
+    - `actual_difference` to 0.25
+    - `actual_difference` to 1
+    - `variability` to 0.5
+    - `variability` to 3
+
+## Visualization as key step in analysis
+
+Visualizing your data before, during, and after doing statistics or modelling
+your data is an incredibly important and sometimes overlooked aspect of data
+analysis. Sometimes statistics and modelling give you an answer that you sort of
+expect, but the actual data is telling a different story.
+
+#### Challenge
+
+There are several datasets found within the `datasauRus` package. This package
+is trying to illustrate a point. First, let's look at the mean and standard
+deviation of the variables in each dataset:
+
+```{r}
+datasaurus_dozen %>%
+    group_by(dataset) %>%
+    summarise(
+        mean_x = mean(x),
+        sd_x = sd(x),
+        mean_y = mean(y),
+        sd_y = sd(y)
+    )
+```
+
+They're all basically the same... What about for linear regression? (`lm` is the
+same as `glm` with family `gaussian`.)
+
+```{r}
+# Linear regression on each dataset
+datasaurus_dozen %>% 
+    group_by(dataset) %>% 
+    do(tidy(lm(y ~ x, data = .))[2, ])
+```
+
+Same thing, all are more or less the same. But, there is something seriously
+wrong with every since one of the datasets. Pair up and explore the data
+visually to see what's going on. Can you see it?
+
+```{r, echo=FALSE, eval=FALSE}
+# Solution
+ggplot(datasaurus_dozen, aes(x = x, y = y, colour = dataset)) +
+    geom_point() +
+    theme_void() +
+    theme(legend.position = "none") +
+    facet_wrap(~ dataset, ncol = 3)
+```
+
+### Other examples of using visualization to find a problem
+
+From what you found in the challenge above, can you see why it's dangerous to
+not visualize the data first? Sometimes just visualizing the data can give you
+greater insight into the data and meaning than any statistical or mathematical
+model can provide. (For a real-world example of this, check out this 
+[article](https://www.wired.com/2014/10/astrophysics-interstellar-black-hole/)
+on the discoveries physicists made when they created a visualization of a black
+hole based on the math when making the movie Interstellar).
+
+But even the type of graph you use can hide data. Check out the figure below for
+an example of that. In general, it's better to visualize the raw data points,
+but this quickly becomes a problem as you get more data.
+
+![Visualizing raw data vs aggregate](boxplot-problem.gif)
```

Comment:
The gif is hypnotizing! I could watch it for hours =) I don't completely understand how it is showing that thhe distributions differ though, do you have a link to where you got it so that I can read more?

Reply:
Oh, shiiittt, right, I forgot about including a citation to it.. oops! I'll add it.

The changes in the distributions show that as the distribution changes, the boxplots do *not* change! So boxplots can be tricky too.